src/Microsoft.Diagnostics.Monitoring.EventPipe/Logs/EventLogsPipeline.cs

```diff
@@ -93,130 +106,163 @@ protected override Task OnEventSourceAvailable(EventPipeEventSource eventSource,
-                string categoryName = (string)traceEvent.PayloadByName("LoggerName");
-                int eventId = (int)traceEvent.PayloadByName("EventId");
-                string eventName = (string)traceEvent.PayloadByName("EventName");
-                string exceptionJson = (string)traceEvent.PayloadByName("ExceptionJson");
-                string argsJson = (string)traceEvent.PayloadByName("ArgumentsJson");
-
-                // There's a bug that causes some of the columns to get mixed up
-                if (eventName.StartsWith("{"))
-                {
-                    argsJson = exceptionJson;
-                    exceptionJson = eventName;
-                    eventName = null;
-                }
-
-                if (string.IsNullOrEmpty(argsJson))
-                {
-                    return;
-                }
-
-                Exception exception = null;
-
-                ILogger logger = _factory.CreateLogger(categoryName);
-                List<IDisposable> scopes = new();
-
-                //
-                // The MessageJson event will occur with an ActivityId equal to the most relevant activity branch and we can backtrack to the root of the tree
-                // to grab all applicable scopes (where each node we visit is an applicable scope).
-                //
-                // Ideally the ActivityId will always exist in our tree, however if another trace is ongoing that is interested in an event start
-                // within the same async context as our log message then there will be nodes+edges that our tree is unaware of.
-                // This is because TplEventSource's TasksFlowActivityIds is a singleton implementation that is shared for all traces,
-                // regardless of if the other traces have TasksFlowActivityIds enabled.
-                //
-                // In this scenario there's still a chance that only a single branch has occurred and we're the first event logged with the newly branched ActivityId.
-                // In which case we can use the RelatedActivityId to find our way back onto the tree.
-                //
-                // If not then we will be operating on a subtree without a way of getting back to the root node and will only have a subset (if any) of the
-                // applicable scopes.
-                //
-                if (activityIdToScope.TryGetValue(traceEvent.ActivityID, out LogScopeItem scopeItem) ||
-                    activityIdToScope.TryGetValue(traceEvent.RelatedActivityID, out scopeItem))
-                {
-                    while (scopeItem != null)
+            if (_logger != null)
+            {
+                eventSource.Dynamic.AddCallbackForProviderEvent(LoggingSourceConfiguration.MicrosoftExtensionsLoggingProviderName, "MessageJson", (traceEvent) => {
+                    if (!activityIdToScope.TryGetValue(traceEvent.ActivityID, out LogScopeItem scopeItem))
                     {
```

Comment:
I'm assuming you kept the two code paths separate because ultimately we will remove the _logger == null path. However, I'm assuming there's a lot of overlap between our reading of the log events and the new code.

Reply:
Ya there is some overlap. But would prefer to leave as-is and spend the time updating consumers to use the new pipeline, remove the old one instead of spending time to make the side-by-side really slick. Thoughts?